module/assessments/helper.js

```diff
@@ -422,9 +424,129 @@ module.exports = class AssessmentsHelper {
             ]);
         })
     }
 
+      /**
+   * Assessment meta form.
+   * @method
+   * @name metaForm
+   * @param {String} solutionId - Assessment solution id.
+   * @param {String} subType - can be individual or institutional.
+   * @returns {JSON} - Form details.
+   */
+
+  static metaForm(solutionId,subType) {
+
+    return new Promise(async (resolve, reject) => {
+
+        try {
+
+            let solutionsData = await solutionsHelper.solutionDocuments({
+                _id : solutionId,
+                type : messageConstants.common.ASSESSMENT,
+                subType : subType,
+                isReusable: true
+            }, [
+              messageConstants.common["ASSESSMENT_META_FORM_KEY"]
+            ]);
+
+
+            if (!solutionsData[0]._id) {
+                throw { 
+                    status: httpStatusCode.bad_request.status, 
+                    message: messageConstants.apiResponses.SOLUTION_NOT_FOUND
+                };
+            }
+
+            let filteredData = {
+              "name" : solutionsData[0].assessmentMetaFormKey
+            };
+
+            let observationsMetaForm = 
+            await formsHelper.formDocuments(filteredData,["value"]);
+
+            if( !observationsMetaForm[0] ) {
+              throw { 
+                status: httpStatusCode.bad_request.status, 
+                message: messageConstants.apiResponses.FORM_NOT_FOUND
+              };
+            }
+
+            return resolve({
+                message: messageConstants.apiResponses.OBSERVATION_META_FETCHED,
+                result: observationsMetaForm[0].value
+            });
+
+        } catch (error) {
+            return reject({
+                status: error.status || httpStatusCode.internal_server_error.status,
+                message: error.message || httpStatusCode.internal_server_error.message,
+                errorObject: error
+            });
+        }
+
+    });
+  }
+
+     /**
+     * Create solution and program from assessment template. 
+     * @method
+     * @name create
+     * @param {String} templateId - assessment template id. 
+     * @param {String} userId - Logged in user id.
+     * @param {Object} requestedData - request body data.
+     * @returns {Object} - Create solution from assessment template. 
+     */
+
+    static create( templateId,userId,requestedData,subType ) {
+        return new Promise(async (resolve, reject) => {
+            try {
+  
+              let filterQueryData = {
+                _id : templateId,
+                type : messageConstants.common.ASSESSMENT,
+                subType : subType,
+                isReusable : true
+              }
+
+              let solutionInformation =  {
+                name : requestedData.name,
+                description : requestedData.description
+              };
+
+              if ( requestedData.entities ) {
+                  solutionInformation["entities"] = requestedData.entities;
```

Comment:
@aman-tunerlabs - Validate entity types of the entities being added with the entity type of template id